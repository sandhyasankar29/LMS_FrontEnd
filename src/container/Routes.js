import React from 'react';
import SignIn from './SignIn';
import SignUp from './SignUp';
import { Route, BrowserRouter as Router, Switch } from 'react-router-dom';
import Welcome from './Welcome';
import StudentDashboard from './StudentDashboard';
import AdminTeacherVerify from './AdminTeacherVerify';
import AdminSignIn from './AdminSignIn';
import { history } from './history';
import ForgotPassword from './ForgotPassword';
import TeacherSignIn from './TeacherSignIn';
import TeacherRegister from './TeacherRegister';
import TeacherDashboard from './TeacherDashboard';
import TeacherPasswordChange from './TeacherPasswordChange';
import StudentPastApplication from './StudentPastApplication';
import RevokeAccess from './RevokeAccess';
function Routes() {
    document.body.style.overflow = "hidden"
    return (
        <Router history={history}>
            <div>
                <div>
                    <Switch>
                        <Route path="/" component={Welcome} exact={true} />
                        <Route path="/signin" component={SignIn} exact={true} />
                        <Route path="/studentDashboard" component={StudentDashboard} exact={true} />
                        <Route path="/signup" component={SignUp} exact={true} />
                        <Route path='/adminsignin' component={AdminSignIn} exact={true} />
                        <Route path='/adminteacherverify' component={AdminTeacherVerify} exact={true} />
                        <Route path='/adminrevoke' component={RevokeAccess} exact={true} />
                        <Route path='/forgotpassword' component={ForgotPassword} exact={true} />
                        <Route path='/teachersignin' component={TeacherSignIn} exact={true} />
                        <Route path='/teacherregister' component={TeacherRegister} exact={true} />
                        <Route path='/teacherdashboard' component={TeacherDashboard} exact={true} />
                        <Route path='/teacherchangepassword' component={TeacherPasswordChange} exact={true} />
                        <Route path='/studentpastapplication' component={StudentPastApplication} exact={true} />
                    </Switch>
                </div>
            </div>
        </Router>
    );
}

export default Routes;