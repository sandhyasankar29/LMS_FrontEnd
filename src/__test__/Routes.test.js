import React from 'react';
import ReactDOM from 'react-dom';
import Routes from '../container/Routes';
import { cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'

afterEach(cleanup)
it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Routes></Routes>, div)
})

it("matches snapshot", () => {
    const tree = renderer.create(<Routes></Routes>).toJSON();
    expect(tree).toMatchSnapshot();
})
