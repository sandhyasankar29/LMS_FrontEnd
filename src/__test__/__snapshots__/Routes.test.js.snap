// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div>
    <div
      className="StyledGrommet-sc-19lkkz7-0 kePwwg"
    >
      <div
        className="StyledBox-sc-13pk1d4-0 hLASUq"
      >
        <div
          className="StyledStack-ajspsk-0 fDjhAR"
          onKeyDown={[Function]}
        >
          <div
            className="StyledStack__StyledStackLayer-ajspsk-1 jeVJjk"
          >
            <div
              className="StyledBox-sc-13pk1d4-0 ceQwbc"
            >
              <div
                className="StyledBox-sc-13pk1d4-0 dYebPD"
              >
                <div
                  className="StyledBox-sc-13pk1d4-0 blPiMm"
                >
                  <img
                    className="StyledImage-ey4zx9-0 inAlaK"
                    src="carousal_1.jpg"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            className="StyledStack__StyledStackLayer-ajspsk-1 gBThXn"
          >
            <div
              className="StyledBox-sc-13pk1d4-0 ceQwbc"
            >
              <div
                className="StyledBox-sc-13pk1d4-0 csgqhr"
              >
                <div
                  className="StyledBox-sc-13pk1d4-0 eQhJgJ"
                >
                  <img
                    className="StyledImage-ey4zx9-0 inAlaK"
                    src="carousal_3.jpg"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            className="StyledStack__StyledStackLayer-ajspsk-1 gBThXn"
          >
            <div
              className="StyledBox-sc-13pk1d4-0 dwjKg"
              onBlur={[Function]}
              onFocus={[Function]}
              tabIndex="0"
            />
          </div>
        </div>
      </div>
      <div
        className="StyledBox-sc-13pk1d4-0 gmTcRP"
      >
        <div
          className="StyledGrid-sc-1wofa1l-0 bPtncB"
        >
          <div
            className="StyledBox-sc-13pk1d4-0 eYIiIW"
            id="student"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            style={
              Object {
                "marginTop": "-2%",
              }
            }
            tabIndex={0}
          >
            <div
              className="StyledBox-sc-13pk1d4-0 ZZZWw"
            >
              <div
                className="StyledBox-sc-13pk1d4-0 kojPrO"
              >
                <svg
                  aria-label="Book"
                  className="StyledIcon-ofa7kd-0 lneDmx"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10,1 L10,11 L13,9 L16,11 L16,1 M5.5,18 C4.11928813,18 3,19.1192881 3,20.5 C3,21.8807119 4.11928813,23 5.5,23 L22,23 M3,20.5 L3,3.5 C3,2.11928813 4.11928813,1 5.5,1 L21,1 L21,18.0073514 L5.49217286,18.0073514 M20.5,18 C19.1192881,18 18,19.1192881 18,20.5 C18,21.8807119 19.1192881,23 20.5,23 L20.5,23"
                    fill="none"
                    stroke="#000"
                    strokeWidth="2"
                  />
                </svg>
                <div
                  className="StyledBox__StyledBoxGap-sc-13pk1d4-1 ggstca"
                />
                <div
                  className="StyledBox-sc-13pk1d4-0 dYebPD"
                >
                  <span
                    className="StyledText-sc-1sadyjn-0 eNnSXE"
                  >
                    Student Portal Login
                  </span>
                  <span
                    className="StyledText-sc-1sadyjn-0 gOVGlS"
                  />
                </div>
              </div>
            </div>
            <footer
              className="StyledBox-sc-13pk1d4-0 hauLXg"
            >
              <span
                className="StyledText-sc-1sadyjn-0 gJdaHr"
              >
                Student Potal to Apply OD/ML
              </span>
            </footer>
          </div>
          <div
            className="StyledBox-sc-13pk1d4-0 eYIiIW"
            id="admin"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            style={
              Object {
                "marginTop": "-2%",
              }
            }
            tabIndex={0}
          >
            <div
              className="StyledBox-sc-13pk1d4-0 ZZZWw"
            >
              <div
                className="StyledBox-sc-13pk1d4-0 kojPrO"
              >
                <svg
                  aria-label="Storage"
                  className="StyledIcon-ofa7kd-0 lneDmx"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M2,5.07692308 C2,5.07692308 3.66666667,2 12,2 C20.3333333,2 22,5.07692308 22,5.07692308 L22,18.9230769 C22,18.9230769 20.3333333,22 12,22 C3.66666667,22 2,18.9230769 2,18.9230769 L2,5.07692308 Z M2,13 C2,13 5.33333333,16 12,16 C18.6666667,16 22,13 22,13 M2,7 C2,7 5.33333333,10 12,10 C18.6666667,10 22,7 22,7"
                    fill="none"
                    stroke="#000000"
                    strokeWidth="2"
                  />
                </svg>
                <div
                  className="StyledBox__StyledBoxGap-sc-13pk1d4-1 ggstca"
                />
                <div
                  className="StyledBox-sc-13pk1d4-0 dYebPD"
                >
                  <span
                    className="StyledText-sc-1sadyjn-0 eNnSXE"
                  >
                    Admin Portal Login
                  </span>
                  <span
                    className="StyledText-sc-1sadyjn-0 gOVGlS"
                  />
                </div>
              </div>
            </div>
            <footer
              className="StyledBox-sc-13pk1d4-0 hauLXg"
            >
              <span
                className="StyledText-sc-1sadyjn-0 gJdaHr"
              >
                Admin Portal to Manage Records
              </span>
            </footer>
          </div>
          <div
            className="StyledBox-sc-13pk1d4-0 eYIiIW"
            id="teacher"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            style={
              Object {
                "marginTop": "-2%",
              }
            }
            tabIndex={0}
          >
            <div
              className="StyledBox-sc-13pk1d4-0 ZZZWw"
            >
              <div
                className="StyledBox-sc-13pk1d4-0 kojPrO"
              >
                <svg
                  aria-label="Group"
                  className="StyledIcon-ofa7kd-0 lneDmx"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12,13 C14.209139,13 16,11.209139 16,9 C16,6.790861 14.209139,5 12,5 C9.790861,5 8,6.790861 8,9 C8,11.209139 9.790861,13 12,13 Z M6,22 L6,19 C6,15.6862915 8.6862915,13 12,13 C15.3137085,13 18,15.6862915 18,19 L18,22 M13,5 C13.4037285,3.33566165 15.0151447,2 17,2 C19.172216,2 20.98052,3.790861 21,6 C20.98052,8.209139 19.172216,10 17,10 L16,10 L17,10 C20.287544,10 23,12.6862915 23,16 L23,18 M11,5 C10.5962715,3.33566165 8.98485529,2 7,2 C4.82778404,2 3.01948003,3.790861 3,6 C3.01948003,8.209139 4.82778404,10 7,10 L8,10 L7,10 C3.71245602,10 1,12.6862915 1,16 L1,18"
                    fill="none"
                    stroke="#000"
                    strokeWidth="2"
                  />
                </svg>
                <div
                  className="StyledBox__StyledBoxGap-sc-13pk1d4-1 ggstca"
                />
                <div
                  className="StyledBox-sc-13pk1d4-0 dYebPD"
                >
                  <span
                    className="StyledText-sc-1sadyjn-0 eNnSXE"
                  >
                    Teacher Portal
                  </span>
                  <span
                    className="StyledText-sc-1sadyjn-0 gOVGlS"
                  />
                </div>
              </div>
            </div>
            <footer
              className="StyledBox-sc-13pk1d4-0 hauLXg"
            >
              <span
                className="StyledText-sc-1sadyjn-0 gJdaHr"
              >
                Teacher Portal to Approve Student Records
              </span>
            </footer>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
